osf/management/commands/rebrand_preprint_provider.py

```diff
@@ -0,0 +1,108 @@
+"""
+This script is destructive and should be used with caution.
+Before use, a new provider must already be created with correct subjects and assets.
+
+A change to the nginx config, redirecting "/preprints/<src_id>/*" to "/preprints/<dst_id>/*"
+must go out when this is ran.
+"""
+from __future__ import unicode_literals
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import Count
+import progressbar
+
+from scripts import utils as script_utils
+from osf.management.commands.reindex_preprint_provider import reindex_provider
+from osf.models import PreprintProvider
+from osf.utils.migrations import disable_auto_now_fields
+
+logger = logging.getLogger(__name__)
+
+def rebrand_provider(src_id, dst_id):
```

Comment:
probably not an issue for the specific provider we're looking to change, but one provider has its own doi client?  What do we send when we're creating a DOI for a preprint, and is it something that's going to have to change?

Reply:
The links sent to doi providers should be of one of two forms: `osf.io/<guid>`, or `osf.io/preprints/<provider>/<guid>`. Internal redirection should take care of the former, nginx redirection the latter.